Item 9584: ManageSampleStatusesPanel for sample statuses CRUD within a given container

cnathe opened the pull request:
#### Rationale
See related PRs for further details. We removed the state type field from the core.DataStates insert and update form so this PR changes some existing tests to use the Java API to insert sample status rows instead of the UI.

#### Related Pull Requests
* https://github.com/LabKey/labkey-ui-components/pull/653
* https://github.com/LabKey/platform/pull/2698
* https://github.com/LabKey/sampleManagement/pull/726
* https://github.com/LabKey/biologics/pull/1026
* https://github.com/LabKey/inventory/pull/323

#### Changes
* SampleTypeHelper update to insertSampleState() to use InsertRowsCommand

cnathe committed:
Fix selenium tests to insert sample statuses via API
- we removed the state type field from the core.DataStates insert and update form

cnathe committed:
Add some logging for the experimental feature change to triage SampleStatusTest failure on TC

labkey-susanh commented on src/org/labkey/test/util/SampleTypeHelper.java:
```diff
@@ -322,29 +327,31 @@ public static Boolean setSampleStatusEnabled(boolean enabled)
             {
-                drt.clickInsertNewRow();
-                addSampleState(statePair.getKey(), statePair.getValue().name());
+                insertSampleState(folderPath, statePair.getKey(), statePair.getValue().name());
```
If this is just using the API, is there a need to continue navigating to the data region and all that? The `insertSampleState` helper could also do all the rows in one insert command.

cnathe replied:
Good catch. Fixed / refactored accordingly.

labkey-susanh reviewed (approved):
Seems functional, but might be more efficient.

cnathe committed:
Update usage of SsampleTypeHelper.addSampleStates to not navigate in UI

cnathe committed:
Move sample status test cases to LKSSampleStatusTest in SM module
- since the SM module now is the one that "enables" the sample status feature checks

cnathe committed:
Merge branch 'develop' into fb_smManageStatuses
